olive/passes/onnx/bnb_quantization.py

```diff
@@ -0,0 +1,125 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import logging
+import re
+from typing import Any, Dict, List
+
+import onnx
+from onnxruntime import __version__ as OrtVersion
+from packaging import version
+
+from olive.hardware import AcceleratorSpec
+from olive.model import ONNXModel
+from olive.passes import Pass
+from olive.passes.onnx.common import get_external_data_config, model_proto_to_olive_model
+from olive.passes.pass_config import PassConfigParam
+
+logger = logging.getLogger(__name__)
+
+
+class OnnxBnb4Quantization(Pass):
+    """Quantize MatMul nodes in ONNX model using 4bit FP4/NF4 quantization."""
+
+    @staticmethod
+    def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
+        config = {
+            "quant_type": PassConfigParam(
+                type_=str,
```

Comment:
Nit: Better to set the default value? Somehow it can help on auto optimization.

Reply:
This is similar to the `model_type`, `num_heads` etc in the transformers optimization pass. No default value is set so that we can check model_attributes if the pass config value is not given.

Reply:
Also, the bnb4 quantization is not really useful by itself as a model compression technique since accuracy is not good. It works best when used with fine-tuned lora adapters from qlora. So, we might not even try this pass in auto optimization.